### PR TITLE
1884 change mocking framework

### PR DESCRIPTION
**Pull request by ymersod**

### CODE REVIEW

**Review by CodingDegen — DISMISSED** [review]

File: domains/emissions/Emissions.API/Tests/Helpers/MockHttpClientFactory.cs [hunk 1]
```diff
@@ -1,42 +1,40 @@
...
-            .Protected()
-            // Setup the PROTECTED method to mock
-            .SetupSequence<Task<HttpResponseMessage>>(
-                "SendAsync",
-                ItExpr.IsAny<HttpRequestMessage>(),
-                ItExpr.IsAny<CancellationToken>()
-            );
+        var handlerMock = Substitute.For<HttpMessageHandler>();
+        var handlerPart = handlerMock.GetType().GetMethod("SendAsync", BindingFlags.NonPublic | BindingFlags.Instance)!
+            .Invoke(handlerMock, new object[] { Arg.Any<HttpRequestMessage>(), Arg.Any<CancellationToken>() });
```
Comment (CodeReaper) [inline domains/emissions/Emissions.API/Tests/Helpers/MockHttpClientFactory.cs:19]:
```suggestion
        var handlerPart = handlerMock.GetType().GetMethod("SendAsync", BindingFlags.NonPublic | BindingFlags.Instance)!.Invoke(handlerMock, new object[] { Arg.Any<HttpRequestMessage>(), Arg.Any<CancellationToken>() });
```

File: domains/emissions/Emissions.API/Tests/Helpers/MockHttpClientFactory.cs [hunk 2]
```diff
@@ -1,42 +1,40 @@
...
-            handlerPart = handlerPart.ReturnsAsync(new HttpResponseMessage()
+
+            var httpResponseMessage = new HttpResponseMessage()
             {
                 StatusCode = HttpStatusCode.OK,
                 Content = contentSequence
-            });
+            };
+            var task = Task.FromResult(httpResponseMessage);
+            response.Enqueue(task);
```
Comment (CodeReaper) [inline domains/emissions/Emissions.API/Tests/Helpers/MockHttpClientFactory.cs:33]:
```suggestion
            response.Enqueue(Task.FromResult(httpResponseMessage));
```

File: domains/measurements/Measurements.API/Tests/Helpers/MockHttpClientFactory.cs [hunk 3]
```diff
@@ -22,25 +22,20 @@ public static HttpClient SetupHttpClientFromFile(string resourceName)
...
+        var httpResponseMessage = new HttpResponseMessage()
+        {
+            StatusCode = HttpStatusCode.OK,
+            Content = new StringContent(serialize),
+        };
+        var task = Task.FromResult(httpResponseMessage);
 
+        handlerMock.GetType().GetMethod("SendAsync", BindingFlags.Instance | BindingFlags.NonPublic)!
+            .Invoke(handlerMock, new object[] { Arg.Any<HttpRequestMessage>(), Arg.Any<CancellationToken>() })
+            .Returns(task);
```
Comment (CodeReaper) [inline domains/measurements/Measurements.API/Tests/Helpers/MockHttpClientFactory.cs:36]:
```suggestion
        handlerMock
            .GetType()
            .GetMethod("SendAsync", BindingFlags.Instance | BindingFlags.NonPublic)!
            .Invoke(handlerMock, new object[] { Arg.Any<HttpRequestMessage>(), Arg.Any<CancellationToken>() })
            .Returns(Task.FromResult(httpResponseMessage));
```

File: domains/measurements/Measurements.API/Tests/Helpers/MockHttpClientFactory.cs [hunk 4]
```diff
@@ -22,25 +22,20 @@ public static HttpClient SetupHttpClientFromFile(string resourceName)
...
-            {
-                StatusCode = HttpStatusCode.OK,
-                Content = new StringContent(serialize),
-            }).Verifiable();
+        var httpResponseMessage = new HttpResponseMessage()
+        {
+            StatusCode = HttpStatusCode.OK,
+            Content = new StringContent(serialize),
+        };
+        var task = Task.FromResult(httpResponseMessage);
```
Comment (CodeReaper) [inline domains/measurements/Measurements.API/Tests/Helpers/MockHttpClientFactory.cs:32]:
```suggestion
```

Included in another suggestion

**Review by CodeReaper — APPROVED** [review]

**Pull request merged**